### PR TITLE
Editorial: Adding style-guide.md with casing conventions explainer.

### PR DESCRIPTION
**Pull request by sffc**
Follow-up to https://github.com/tc39/proposal-intl-displaynames/issues/32.

### CODE REVIEW

**Comment by sffc** [comment]
@waldemarhorwat @wycats @domenic @gesa @littledan @leobalter 

I would like to adopt this style guide for ECMA-402.  It recommends Option 3 from my presentation in July, to use camel case everywhere.  As @littledan suggested, these guidelines would be scoped only to ECMA-402 and would not restrict what ECMA-262 could do in the future.

I am adding this to the October agenda in https://github.com/tc39/agendas/pull/622.

**Comment by sffc** [comment]
Backref: Fixes #363

File: docs/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            - [Decision](#decision-2)
+    - [Identifiers defined outside ECMA-402](#identifiers-defined-outside-ecma-402)
+        - [Examples](#examples-1)
+        - [Alternative: Convert identifiers to camel case](#alternative-convert-identifiers-to-camel-case)
+            - [Pros](#pros-3)
+            - [Cons](#cons-3)
+            - [Decision](#decision-3)
+
+
+*Table of Contents generated using https://magnetikonline.github.io/markdown-toc-generate/*
```
Comment (leobalter) [inline docs/style-guide.md]: I'm fine adding this, but I wonder if we should move it to the end of the file.

File: docs/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+- It is difficult to take this option seriously since it would add debt to the specification.
+- If strongly desired, the spec could be changed later to allow the kebab-cased strings.
+
+### Identifiers defined outside ECMA-402
+
+:star2: *ECMA-402 sits between ECMA-262 and internationalization standards such as UTS #35. For identifiers whose syntax comes from a different specification, follow the convention in that specification.* :star2:
+
+#### Examples
+
+Locale identifiers are defined in [UTS #35 Section 3](https://www.unicode.org/reports/tr35/#Identifiers).  Examples:
```
Comment (srl295) [inline docs/style-guide.md]: I'd prefer something such as 
> [BCP47](http://www.rfc-editor.org/rfc/bcp/bcp47.txt) identifiers, using registered extensions as described in [UTS#35§3](https://www.unicode.org/reports/tr35/#Identifiers)

The identifiers are and must be valid BCP47.

**Comment by leobalter** [comment]
TC39 approved this document during the meeting in 2019-10-02

**Comment by sffc** [comment]
@waldemarhorwat had some suggestions from TC39; could you leave comments here?  Thanks!

**Comment by leobalter** [comment]
ping @waldemarhorwat

File: docs/style-guide.md (hunk 1 above)
Comment (sffc) [reply docs/style-guide.md]: There isn't really anywhere to put it down there.  I like having it close to the actual ToC so it's easy to find how to update it.

File: docs/style-guide.md (hunk 2 above)
Comment (sffc) [reply docs/style-guide.md]: Done.

**Comment by sffc** [comment]
I'm going to merge this.  Feedback from @waldemarhorwat can be integrated in a follow-up PR.

**Pull request merged**